editor/block-toolbar/index.js

```diff
@@ -29,17 +35,71 @@ class BlockToolbar extends Component {
 	constructor() {
 		super( ...arguments );
 		this.toggleMobileControls = this.toggleMobileControls.bind( this );
+		this.bindNode = this.bindNode.bind( this );
+		this.onKeyUp = this.onKeyUp.bind( this );
+		this.onKeyDown = this.onKeyDown.bind( this );
 		this.state = {
 			showMobileControls: false,
 		};
 	}
 
+	componentDidMount() {
+		document.addEventListener( 'keyup', this.onKeyUp );
+	}
+
+	componentWillUnmout() {
+		document.removeEventListener( 'keyup', this.onKeyUp );
+	}
+
+	bindNode( ref ) {
+		this.toolbar = ref;
+	}
+
 	toggleMobileControls() {
 		this.setState( ( state ) => ( {
 			showMobileControls: ! state.showMobileControls,
 		} ) );
 	}
 
+	onKeyDown( event ) {
+		// This is required to avoid messing up with the WritingFlow navigation
+		event.stopPropagation();
+	}
+
+	onKeyUp( event ) {
+		// Is there a better way to focus the selected block
+		const selectedBlock = document.querySelector( '.editor-visual-editor__block.is-selected' );
```

Comment:
Tried avoiding the selector by relying on state, but we do not have a reliable way to focus a block with the state only. The focus block action doesn't refocus if it's already selected.

Reply:
Hm, but this also fires on every keyup?